Sources/Plasma/NucleusLib/inc/hsResMgr.h

```diff
@@ -145,6 +145,9 @@ class hsgResMgr
 public:
     static hsResMgr* ResMgr() { return (hsResMgr*)fResMgr; }
 
+    // External modifier use only
+    static void SetTheResMgr(hsResMgr* mgr) { fResMgr = mgr; }
```

Comment:
This is for ModDLL stuff I assume? Does this actually work in any useful way? I would have expected that the singleton ResMgr object gets passed around and stored in some places that won't be updated by this call.

Reply:
This is used by ModDLL stuff, but for setting the ResMgr _in_ the DLL to the existing instance in the host application. It matches how the plFactory and plTimerCallbackMgr are handled.

The ModDLL stuff does work, and after these changes you can actually define external creatable classes and use them... whether that ends up being useful or not is up for debate 🤷🏼‍♂️

Reply:
Oh, I see, the DLL has its own copy of the variable that stores the singleton ResMgr. I assumed that it would share the variable with the main executable and that this would *replace* the existing ResMgr - but the variable is static, so of course it isn't shared that way. Makes sense now, thanks.